tests/rptest/tests/cluster_config_test.py

```diff
@@ -1614,6 +1614,36 @@ def test_validate_cloud_storage_api_endpoint(self, value, valid):
             #Valid api endpoint
             assert valid
 
+    # None for pct_value is std::nullopt, which defaults to 0.0 in the cloud cache.
+    @cluster(num_nodes=3)
+    @parametrize(size_value=0, pct_value=None, valid=False)
+    @parametrize(size_value=0, pct_value=0.0, valid=False)
+    @parametrize(size_value=0, pct_value=-1.0, valid=False)
+    @parametrize(size_value=0, pct_value=101.0, valid=False)
+    @parametrize(size_value=-1, pct_value=None, valid=False)
+    @parametrize(size_value=1024, pct_value=None, valid=True)
+    @parametrize(size_value=10, pct_value=50.0, valid=True)
+    @parametrize(size_value=0, pct_value=0.1, valid=True)
+    def test_validate_cloud_storage_cache_size_config(self, size_value,
```

Comment:
nit: this seems like a lot of docker containers for a pretty small test. Wondering if it makes sense to reduce the ducktape testing to just one valid and one invalid case, and then unit test the rest in c++?

Reply:
Sounds good, I can reduce the number of nodes down to 1.

I can cover the some of these cases in a unit test for `cloud_storage::cache::validate_cache_config()`, but if what we are trying to assert is end to end cluster configuration validation (since ultimately we are trying to prevent end users from misconfiguring the cluster using the admin endpoint), it feels like ducktape tests are the best place for it?

Let me know what you think.

Reply:
It takes one or two cases to demonstrate/test how invalid inputs looks end-to-end with ducktape. Especially since there isn't any complex transformation happening in between the admin endpoint and the call to the validation code. A unit test seems like a better place to be exhaustive, especially when the endpoint-facing results are the same for each bad validation.

Alternatively maybe this could just be a loop over valid and invalid inputs instead of parameterizing. My point is that it seems expensive to run through these cases, spinning up new containers for each case.

Reply:
Fair enough. Altered test to iterate over test cases instead of parameterizing the test.

Reply:
Thanks!